### DISCUSSION TITLE
How to merge table cells (horizontally and vertically)?

### DISCUSSION THREAD

**Discussion by ptram** [quarto-dev/quarto-cli#8427] [Q&A]
### Description

Hi,

Is there a way to merge table cells, both horizontally and vertically? That is in cases where a cell's content can span across multiple cells, maybe because it gives the same answer to multiple questions.

I've tried with using a double pipe (||) for merging cells horizontally, or the circumflex accent (^) for merging vertically, but it doesn't render as expected.

Is this possible? If not, are there suggested workarounds?

**Comment by cscheid** [comment]
No, rowspans and colspans are not supported in simple markdown tables. With that said, Quarto understands HTML tables: https://quarto.org/docs/authoring/tables.html#html-tables This might work well enough for your case.

**Comment by mcanouil** [comment]
There is Pandoc grid tables, see <https://pandoc.org/chunkedhtml-demo/8.9-tables.html>.
Note that grid table are not supported by all markdown readers.
For instance, the below table will work for HTML, but not in GitHub (gfm format).

```md
+---------------------+-----------------------+
| Location            | Temperature 1961-1990 |
|                     | in degree Celsius     |
|                     +-------+-------+-------+
|                     | min   | mean  | max   |
+=====================+=======+=======+=======+
| Antarctica          | -89.2 | 19.8          |
+---------------------+-------+-------+-------+
| Earth               | -89.2 | 14    | 56.7  |
+---------------------+-------+-------+-------+
```

**Comment by ptram** [comment]
Thank you very much, @cscheid and @mcanouil , for your answers.

I have to confess that I'm not particularly satisfied by the current tables supported by Quarto. Please let me explain why.

- The simplest one is easy and nice, but is conceived for an ideal, bare bone table of the type:

```
Fruit | Q.ty
:----|:----: 
pears | 6
apples | 5
```

It is most likely that a real table looks something like this:

```
Parameter | Meaning | Values
:----------|:---------|:------:
Attack Rate | The speed at which the sound goes from silence to the maximum intensity level. | 0 ~ 10 s
Frequency Cut | Frequency at which the low-pass filter starts to attenuate the harmonic content of the sound. | 300 Hz ~ 10 kHz
```

All considered, I could live with it, in particular if tabs are accepted and can help create some sort of internal alignment.

Unfortunately, Quarto can't merge cells (using the syntax I saw proposed for CommonMark, using the "^" and "||" symbols).

- The grid table seems to me to go against the scope of Markdown, that is separating content and formatting. It's an attempt to recreate the old semi-graphical formatting of the dot-matrix printer age.

The worse problem, with this format, is that the content is broken into separate lines, and can no longer be parsed by language tools (like CAT tools, that can identify an unit by the ending period).

- HTML tables reintroduce the complexity and lack of readability that Markdown tries to avoid. They obviously work perfectly fine, but break the Markdown workflow. Technical and scientific works, that are the ones most likely using Markdown, are full of complex tables.

- The format I find the easier to deal with is the list table. Quarto is not natively supporting it. In the past I was suggested to compile a filter for it, but I have to admit that I'm totally unable to do it.

The advantage of this format is that complex tables are perfectly readable. Everything is set in a symmetric disposition. It doesn't look like a table, but Markdown is meant to deal with content, not appearance. So, I would appreciate it for its scope.

Paolo

**Reply by mcanouil** [reply]
Note that most of the parsing is done by Pandoc.
You could make a feature request to Pandoc to support other table format as Pandoc extension (which are grid tables, etc.).

If this comes to Pandoc, then it will come to Quarto.

**Reply by cscheid** [reply]
> I have to confess that I'm not particularly satisfied by the current tables supported by Quarto

Thanks for the feedback.

**Comment by ptram** [comment]
Just as a followup: I could add the list-table filter developed by Martin Fisher.

https://github.com/pandoc-ext/list-table

I'm not totally sure this is documented in the excellent Quarto documentation, but I just had to put the `list-table.lua` file inside my project folder. Then, I had to add this section to the `_quarto.yml` setting file, just after the `type` section:

```
filters:
  - list-table.lua
```

Quarto rendered my list table without a hassle. Beautiful!

**Reply by mcanouil** [reply]
Thanks for sharing the filter.

To note, the use of filters is documented in: <https://quarto.org/docs/extensions/filters.html>.
